internal/controller/controllers/drainer.go

```diff
@@ -0,0 +1,24 @@
+package controllers
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/kubectl/pkg/drain"
+)
+
+//go:generate mockgen --build_flags=--mod=mod -package=controllers -destination=mock_drainer.go . Drainer
+type Drainer interface {
+	RunCordonOrUncordon(helper *drain.Helper, node *corev1.Node, desired bool) error
+	RunNodeDrain(helper *drain.Helper, nodeName string) error
+}
+
+type KubectlDrainer struct{}
+
+var _ Drainer = &KubectlDrainer{}
+
+func (d *KubectlDrainer) RunCordonOrUncordon(helper *drain.Helper, node *corev1.Node, desired bool) error {
+	return drain.RunCordonOrUncordon(helper, node, desired)
+}
+
+func (d *KubectlDrainer) RunNodeDrain(helper *drain.Helper, nodeName string) error {
```

Comment:
What will happened on timeout?

Reply:
Yes, [we configure a 20s timeout for each drain attempt](https://github.com/openshift/assisted-service/pull/5110/files#diff-86415614c6ed40c56c5ca2b5f751a65746ef5c83b61e6f021a4f5d817e4425baR1698) and the user can configure a global timeout after which we'll just move on to deprovisioning.

Reply:
Sorry i missed the bmh fille